lms/djangoapps/completion/management/commands/ComputeCompletion.py

```diff
@@ -0,0 +1,99 @@
+from collections import deque
+
+from django.core.management.base import BaseCommand, CommandError
+
+from opaque_keys.edx.keys import CourseKey
+from xmodule.modulestore.django import modulestore
+from lms.djangoapps.course_api.blocks.api import get_blocks
+from openedx.core.lib.celery.task_utils import emulate_http_request
+from openedx.core.djangoapps.site_configuration.tests.factories import SiteFactory
+from student.tests.factories import UserFactory
+
+class Command(BaseCommand):
+
+    def handle(self, *args, **options):
+        course_name = 'course-v1:wert+qw4tqw4t+qertqert'
+        course_usage_key = self._create_course_usage_key(course_name)
+
+        user = UserFactory.create(
+            username="staff",
+            email="staff@example.com"
+            )
+        site = SiteFactory.create()
+
+        with emulate_http_request(site, user) as request:
+            import cProfile
+
+            api_raw_course_data = get_blocks(
+                request, 
+                course_usage_key,
+                user=user,
+                requested_fields=['completion', 'children']
+            )
+            name_of_course_root = api_raw_course_data['root']
+            course_blocks = api_raw_course_data['blocks']
+
+            print "== get_blocks =="
+            cProfile.runctx('get_blocks(request, course_usage_key, user=user, requested_fields=[\'completion\', \'children\'])', globals(), locals())
```

Comment:
Should we run a profile that doesn't include `completion` as a requested field and compare it to this one?